dkms.in

```diff
@@ -1381,6 +1380,12 @@ do_build()
 {
     [[ $kernelver && $arch ]] || die 16 "do_build: Empty \$kernelver or \$arch"
 
+    # Source build environment file if specified
+    if [[ -n "$build_environment" && -f "$build_environment" ]]; then
+        # shellcheck disable=SC1090
+        source <( (source "$build_environment"; export -p) )
+    fi
```

Comment:
1. ```source <( (source "$build_environment" &>/dev/null; export -p) )``` will be better.
2. Since the CC and LD variables will be set later in do_build(), this block should be moved to after that point.